src/sentry/api/endpoints/organization_repository_details.py

```diff
@@ -109,7 +106,9 @@ def delete(self, request, organization, repo_id):
 
             countdown = 3600 if has_commits else 0
 
-            repo.rename_on_pending_deletion()
+            with transaction.atomic():
```

Comment:
@mattrobenolt let me know if this is a better solution

Reply:
No, we shouldn't change the way it was.

I'd prefer that `rename_on_pending_deletion()` just didn't call `.save()` or only saves the one field it changes instead of updating the entire row. But I haven't looked into this code enough.

Also not sure why you'd wrap this in atomic here. That wouldn't do anything really.

Feel free to grab me if you wanna chat about this a bit more, but we definitely need to restore the older behavior of `.filter().update()` instead of what you have here since that changes the behavior quite a bit.